frontend/src/types/fetch-state.ts

```diff
@@ -1,6 +1,21 @@
-export interface FetchState {
+import type { NuxtError } from "@nuxt/types"
+
+export interface BaseFetchState {
   isFetching: boolean
-  fetchingError: null | string
   hasStarted?: boolean
   isFinished?: boolean
 }
+
+export interface FetchState extends BaseFetchState {
+  fetchingError: null | string
+}
+
+/**
+ * The fetching error has the props required for Nuxt's
+ * error page.
+ * TODO: replace FetchState with FetchStateWithNuxtError
+ * in the provider, main media and related media stores.
+ */
+export interface FetchStateWithNuxtError extends BaseFetchState {
+  fetchingError: null | NuxtError
+}
```

Comment:
To avoid the verbose name, we could make `FetchState` generic:

```suggestion
export interface FetchState<ErrorType = string> extends BaseFetchState {
  fetchingError: null | ErrorType
}
```

Then everywhere using `FetchStateWithNuxtError` can just use `FetchState<NuxtError>`. Once all usages have been converted to pass `NuxtError` we could change the default to `NuxtError` instead of `string` and remove the explicit generic parameters _or_ just remove the generic capability altogether.